src/comps/recover.tsx

```diff
@@ -0,0 +1,233 @@
+      throw new Error(err);
+    }
+  };
+  const generateScripts = async () => {
+    try {
+      const stxAddress = searchParams.get("stxAddress");
+      const txId = searchParams.get("txId");
+
+      // ensure the user is signed in before proceeding
+      if (!walletInfo.selectedWallet) {
+        notify({
+          type: NotificationStatusType.ERROR,
+          message: "Please connect your wallet to proceed",
+        });
+        throw new Error("User not connected");
+      }
+
+      // ensure that the stx address and txId are present
+      if (!stxAddress || !txId) {
+        notify({
+          type: NotificationStatusType.ERROR,
+          message: "Invalid URL parameters",
+        });
+        throw new Error("User not connected");
+      }
+
+      const signersAggregatePubKey = (await getAggregateKey()).slice(2);
+
+      // Combine the version and hash into a single Uint8Array
+      const serializedAddress = serializeCVBytes(principalCV(stxAddress));
+
+      // get the publicKey from the user payment address
+      // user cannot continue if they're not connected
+      const paymentAddress = walletInfo.addresses.payment!;
+
+      const reclaimPublicKey = paymentAddress.publicKey;
+
+      // Parse lockTime from env variable
+      const parsedLockTime = parseInt(lockTime || "144");
+
+      // Create the reclaim script and convert to Buffer
+      const reclaimScript = Buffer.from(
+        createReclaimScript(parsedLockTime, reclaimPublicKey),
+      );
+
+      const reclaimScriptHex = uint8ArrayToHexString(reclaimScript);
+
+      const signerUint8Array = hexToUint8Array(signersAggregatePubKey!);
+
+      const depositScript = Buffer.from(
+        createDepositScript(signerUint8Array, maxFee, serializedAddress),
+      );
+      // convert buffer to hex
+      const depositScriptHexPreHash = uint8ArrayToHexString(depositScript);
+
+      const emilyReqPayload = {
+        bitcoinTxid: txId,
+        bitcoinTxOutputIndex: 0,
+        reclaimScript: reclaimScriptHex,
+        depositScript: depositScriptHexPreHash,
+      };
+
+      console.log({ emilyReqPayload });
+      //console.log({ emilyReqPayloadClient: JSON.stringify(emilyReqPayload) });
+
+      // make emily post request
+      const response = await fetch("/api/emilyDeposit", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(emilyReqPayload),
+      });
+
+      if (!response.ok) {
+        notify({
+          type: NotificationStatusType.ERROR,
+          message: `Issue with Request to Emily`,
+        });
+
+        throw new Error("Error with the request");
+      }
+      notify({
+        type: NotificationStatusType.SUCCESS,
+        message: `Request successful`,
+      });
+      setShowStepper(true);
+    } catch (err: any) {
+      throw new Error(err);
+    }
+  };
+
+  const renderTxId = () => {
+    const txId = searchParams.get("txId") || "";
```

Comment:
Given the funny behaviour in chrome, should we switch to lowercase param?

Reply:
Added a check to ensure it get from both cases